README.md

```diff
@@ -1,2 +1,47 @@
-# static-site-hosting
-This repository is a demostration of various options to host static websites.
+# Guide to Hosting Static Websites
+
+Hosting static websites, whether it's documentation or a React website, is a common practice. There are numerous options
+available, both paid and free, catering to different levels of load and technical expertise.
+
+This guide aims to compare several hosting options, providing examples and working templates to help you make an
+informed decision. It begins with a general introduction to static website hosting and covers the following options:
+
+- **GitHub Pages**: Host your static website directly from a GitHub repository.
+- **AWS S3**: Utilize Amazon S3 for reliable and scalable static website hosting.
+- **Azure Storage**: Leverage Azure Storage to deploy and manage your static website.
+- **Google Cloud Web Hosting**: Deploy your static website on the Google Cloud platform.
+- **VM Hosting**: Set up a virtual machine to host your static website.
+- **Netlify**: Explore the powerful features of Netlify for static website hosting and continuous deployment.
+- **Cloudflare**: Discover how Cloudflare can enhance your static website's performance, security, and availability.
+
+This repository contains the source of this guide as well as the scripts for implementing the various solutions
+discussed.
+
+Given the ever-evolving nature of these options and the multitude of alternatives for hosting static websites, we
+encourage you to contribute to this guide. **Your insights and expertise are valuable in keeping this resource
+up-to-date and comprehensive.**
+
+Thank you for your contributions and happy hosting!
+
+## Development
+
+The site is built using [MkDocs](https://www.mkdocs.org/).
+
+### Prerequisites:
+
+- Python
```

Comment:
Maybe specify the version, as I believe Python minor versions can break tools quite a lot.
On mkdocs site install guide 3.8.2 is mentioned, so we maybe could say that this is the lowest it can go: https://www.mkdocs.org/user-guide/installation/

Reply:
Good point. Updated.